protocols/autonat/src/v1/behaviour/as_server.rs

```diff
@@ -107,6 +107,27 @@ impl<'a> HandleInnerEvent for AsServer<'a> {
                     },
             } => {
                 let probe_id = self.probe_id.next();
+                if self.connected.get(&peer).is_none() {
+                    tracing::debug!(
+                        %peer,
+                        "Reject inbound dial request from peer since it is not connected"
+                    );
+
+                    let response = DialResponse {
+                        result: Err(ResponseError::DialRefused),
+                        status_text: Some("dial from non-connected peer".to_string()),
+                    };
+                    let _ = self.inner.send_response(channel, response);
```

Comment:
Do we want to send a response to a disconnected peer?

Reply:
I also had doubts whether this should be done or not. But after giving it some more thoughts I also don't think it is necessary. Also because we inform the Swarm that the dial has been refused later on. I'll remove it.